Edit three Setup guides for Cloud users

ptgott opened the pull request:
Teleport Daemon

- Use a Notice for a warning re: the Auth Service scoped to
  self-hosted users
- Misc grammar/style/clarity tweaks

Move the Enterprise License File page

The page does not belong in the Setup section, since it does not
apply to Cloud and OSS users. I have moved the page into the
Enterprise section.

Networking guide

- Add tabbed instructions so users who have selected a given scope
  don't see content for other scopes.

- Mention auth_service.proxy_listener_mode instead of
  tls_routing_enabled for self-hosted readers

- Misc grammar, style, and clarity edits

Scaling

  We already added an edition warning in a Details box, but I thought
  a Notice would be more prominent. I've also added some minor grammar,
  style, and clarity tweaks.

ptgott committed:
Edit three Setup guides for Cloud users

Teleport Daemon

- Use a Notice for a warning re: the Auth Service scoped to
  self-hosted users
- Misc grammar/style/clarity tweaks

Move the Enterprise License File page

The page does not belong in the Setup section, since it does not
apply to Cloud and OSS users. I have moved the page into the
Enterprise section.

Networking guide

- Add tabbed instructions so users who have selected a given scope
  don't see content for other scopes.

- Mention auth_service.proxy_listener_mode instead of
  tls_routing_enabled for self-hosted readers

- Misc grammar, style, and clarity edits

Scaling

  We already added an edition warning in a Details box, but I thought
  a Notice would be more prominent. I've also added some minor grammar,
  style, and clarity tweaks.

zmb3 commented on docs/pages/setup/reference/networking.mdx:
```diff
@@ -5,9 +5,13 @@ description: Teleport ports and networking requirements
+<Tabs>
+<TabItem scope={["oss", "enterprise"]} label="Self-Hosted">
+
+All Teleport services (the Proxy Service, Auth Service, and Nodes) have an
```
app service, db service, windows desktop service all have `public_addr` as well.

ptgott replied:
At some point, I started conflating the term "Node" with "all Teleport services that aren't the Auth/Proxy." As you point out, this is incorrect, and our architecture overview guide [says as much](https://goteleport.com/docs/architecture/overview/#definitions). I still think we need some catch-all term for services that aren't the auth/proxy. I'll create a separate issue for this and fix my usage here.

zmb3 commented on docs/pages/setup/reference/networking.mdx:
```diff
@@ -18,22 +22,51 @@ following use cases:
+</TabItem>
+<TabItem scope={["cloud"]} label="Teleport Cloud">
+
+Teleport Nodes have an optional `public_addr` property that you can modify in
```
Cloud customers likely want to know about the public_addr property for app, db, and desktop access.

zmb3 commented on docs/pages/setup/reference/networking.mdx:
```diff
@@ -18,22 +22,51 @@ following use cases:
+supports HTTP CONNECT tunneling.
+
+To use HTTP CONNECT tunneling, simply set either the `HTTPS_PROXY` or
+`HTTP_PROXY` variables of the `teleport` daemon's shell environment. When the
```
```suggestion
`HTTP_PROXY` variables of the `teleport` daemon's environment. When the
```

Remove `shell` as there is no requirement to run the Teleport daemon via a shell.

zmb3 commented on docs/pages/setup/reference/networking.mdx:
```diff
@@ -18,22 +22,51 @@ following use cases:
 
-To use HTTP CONNECT tunneling, simply set either the `HTTPS_PROXY` or `HTTP_PROXY` environment variables and when Teleport builds and establishes the reverse tunnel to the main cluster, it will funnel all traffic through the proxy. Specifically, if using the default configuration, Teleport will tunnel ports `3024` (SSH, reverse tunnel) and `3080` (HTTPS, establishing trust) through the proxy.
+If using the default configuration, Teleport will tunnel ports `3024` (SSH,
+reverse tunnel) and `3080` (HTTPS, establishing trust) through the proxy.
```
I'm not sure what we're trying to say about port 3080 here. It is not "tunneled through the proxy" - it's just the listen port on the proxy. There is no tunneling taking place on port 3080.

ptgott committed:
Address PR feedback

- Imply that there are non-Auth/Proxy services beside Nodes
- Remove "shell" where it is unnecessary
- Revert the "HTTP CONNECT" section to where it was before this
  change. The copy-edits were inaccurate, but I don't have time
  to perform the edits agin more carefully.

ulysseskan commented on docs/pages/setup/reference/networking.mdx:
```diff
@@ -47,26 +49,17 @@ following use cases:
-
-If using the default configuration, Teleport will tunnel ports `3024` (SSH,
-reverse tunnel) and `3080` (HTTPS, establishing trust) through the proxy.
+To use HTTP CONNECT tunneling, simply set either the `HTTPS_PROXY` or `HTTP_PROXY` environment variables and when Teleport builds and establishes the reverse tunnel to the main cluster, it will funnel all traffic through the proxy. Specifically, if using the default configuration, Teleport will tunnel ports `3024` (SSH, reverse tunnel) and `3080` (HTTPS, establishing trust) through the proxy.
```
Minor, but I would remove the word 'simply'.  Most people reading this will know how to set environment variables but a few might not.  Perhaps include an example like `export HTTPS_PROXY=https://my.proxy:8080`.

ptgott committed:
Address PR feedback

Simplify the HTTP CONNECT section by bringing the example closer
to the explanation of this functionality and removing some
ambiguity over whether one or both of HTTPS_PROXY and HTTP_PROXY are
required. The original text suggested that either is required, then
that both are required. The new text says to assign both, then
explains why.

ulysseskan reviewed (approved):
👍
